plugins/pangea/textGuard.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  HookEventType,
+  PluginContext,
+  PluginHandler,
+  PluginParameters,
+} from '../types';
+import { post, getText } from '../utils';
+import { VERSION } from './version';
+
+export const handler: PluginHandler = async (
+  context: PluginContext,
+  parameters: PluginParameters,
+  eventType: HookEventType,
+  options: {}
+) => {
+  let error = null;
+  let verdict = false;
+  let data = null;
+  try {
+    const text = getText(context, eventType);
+
+    // TODO: Update to v1 once released
+    const url = `https://ai-guard.${parameters.credentials?.domain}/v1beta/text/guard`;
```

Comment:
Can we have a early check for the required fields please? i.e `recipe` and `domain` etc with meaningful errors please?